src/integritysetup/integritysetup.c

```diff
@@ -153,9 +153,13 @@ static int run(int argc, char *argv[]) {
                 }
 
                 if (!isempty(arg_existing_data_device)) {
+#if HAVE_CRYPT_INIT_DATA_DEVICE
                         r = crypt_init_data_device(&cd, device, arg_existing_data_device);
                         if (r < 0)
                                 return log_error_errno(r, "Failed to add separate data device: %m");
+#else
+                        log_warning("data-device= option not supported by current version of libcryptsetup, ignoring.");
```

Comment:
is this really optional? i assumed if the data device is not specified when we are asked to the integrity stuff simply won't work. or what am I missing?

Hence, I would assume this here should be fatal?

Reply:
it's an optional value, there's no fatal error if it's not provided

Reply:
Hmm? so if i have set up my integrity volume with data device x and tags/integrity info on device y on my devel machine, and then want to use that on some other system via /etc/integritytab, and hence specify the two devices x and y in /etc/integritytab there and boot, then this will ignore the data device x and just use the device y, but since that contains no user data, but only integrity info it will fail with an ugly error, no?

I either don't understand what "data-device=" is actually about, or it's not optional at all if you already have a setup that splits up user data from integrity data storage and intend to mount that...

Reply:
I'm just referring to how to the argument parsing is set up. The integritysetup manpage says:

> --data-device
>     Specify a separate data device that contains existing data. The <device> then will contain calculated
>     integrity tags and journal for this data device.

Reply:
Yes even with the cryptsetup command line it's entirely optional, I guess it stores data inline or something:

```
$ sudo integritysetup format /tmp/integrity 

WARNING!
========
This will overwrite data on /tmp/integrity irrevocably.

Are you sure? (Type 'yes' in capital letters): YES
Formatted with tag size 4, internal integrity crc32c.
Wiping device to initialize integrity checksum.
You can interrupt this by pressing CTRL+c (rest of not wiped device will contain invalid checksum).
Finished, time 00:00.026,   23 MiB written, speed 914.4 MiB/s
$ sudo integritysetup open /tmp/integrity foo
$ sudo integritysetup status foo
/dev/mapper/foo is active.
  type:    INTEGRITY
  tag size: 4
  integrity: crc32c
  device:  /dev/loop5
  loop:    /tmp/integrity
  sector size:  512 bytes
  interleave sectors: 32768
  size:    49128 sectors
  mode:    read/write
  failures: 0
  journal size: 180224 bytes
  journal watermark: 50%
  journal commit time: 10000 ms
```

Reply:
If a user is actually using a device to store the checksums and a separate device for the data `--data-device`, the data device isn't optional.  Without it you have no data.  When you do a integrity format, the default is to interleave the checksums with the data on the same device.

Reply:
so it sounds like the parameter is optional, but if it is passed then it needs to be used - I'll update to error out instead

Reply:
never mind, it was wrong to call crypt_init_data_device here, will remove it